### PR TITLE
Feedback for TFT notebook

### PR DESCRIPTION
**Pull request by jdamp-bis**
Additional comments and questions:

**General comments**:

- We should add pydot, keras and tqdm to the requirements
- Personally, I would use Pytorch as a default backend (in particular after realizing how tough it is to set up tensorflow on Windows with GPU support)

**Section 2:**

- Figure 2: Why are we moving df_infl_m to month end?

- df_oil.pct_change().dropna() gives a deprecation warning with pandas >= 2.1 about the handling of nan values (the default option "pad" in pct_change is deprecated). We should check what our desired behavior on nan values is

-  Section 2.4: What is the meaning of the indices s and l?

- Section 2.4, `#| label: "Create batches of data"`: Maybe we can add one line explaining why the folds differ in length?

**Section 3:**
- `# sec-fc:` What is the meaning of the index s?

**Section 4:** 
- Did introducing the gate already improve upon the simple fully connected model? Maybe we can add one line mentioning this, possibly also referring to Figure 8.

**Section 5:**
-  `#| label: tbl-lstm_summary`: Won't this code raise an error when monthly and daily frequency have different amount of features? (due to averaging)
- Where is the "loss" curve in Figure 10a?
- Can we mention Figure 10 in the text and discuss whether the LSTM has improved anything about the performance?

**Section 7:**
- It is not clear to me what we learn from the embedding diagrams

### CODE REVIEW

**Review by dkgaraujo — APPROVED** [review]

**Comment by dkgaraujo** [comment]
> Additional comments and questions:
> 
> **General comments**:
> 
> * We should add pydot, keras and tqdm to the requirements
> * Personally, I would use Pytorch as a default backend (in particular after realizing how tough it is to set up tensorflow on Windows with GPU support)
> 
> **Section 2:**
> 
> * Figure 2: Why are we moving df_infl_m to month end?
> * df_oil.pct_change().dropna() gives a deprecation warning with pandas >= 2.1 about the handling of nan values (the default option "pad" in pct_change is deprecated). We should check what our desired behavior on nan values is
> * Section 2.4: What is the meaning of the indices s and l?
> * Section 2.4, `#| label: "Create batches of data"`: Maybe we can add one line explaining why the folds differ in length?
> 
> **Section 3:**
> 
> * `# sec-fc:` What is the meaning of the index s?
> 
> **Section 4:**
> 
> * Did introducing the gate already improve upon the simple fully connected model? Maybe we can add one line mentioning this, possibly also referring to Figure 8.
> 
> **Section 5:**
> 
> * `#| label: tbl-lstm_summary`: Won't this code raise an error when monthly and daily frequency have different amount of features? (due to averaging)
> * Where is the "loss" curve in Figure 10a?
> * Can we mention Figure 10 in the text and discuss whether the LSTM has improved anything about the performance?
> 
> **Section 7:**
> 
> * It is not clear to me what we learn from the embedding diagrams

All good comments, thanks very much - I'll address them in due time as I go through the file. Thanks for the PR as well.
